### PR TITLE
waiterbot_navigation

### PR DESCRIPTION
**Pull request by mehditlili**

### CODE REVIEW

**Comment by jihoonl** [comment]
@mehditlili  Is there any difference between `turtlebot navigation` and `waiterbot navigation`?

**Comment by jihoonl** [comment]
How about including turtlebot navigation launch files instead of copying?

**Comment by mehditlili** [comment]
Ok I will include them instead, but common_costmap_param.yaml for waiterbot is different from turtlebot as waiterbot includes ir_sensors in the obstacle_layer. I will create a ir_costmap_param where ir_sensor will be added to the obstacle_layer to avoid redundancy

**Comment by mehditlili** [comment]
Waiterbot params are now passed directly to move_base.launch.xml.

File: waiterbot_suite/waiterbot_navigation/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+find_package(catkin REQUIRED)
+
+catkin_package(
+#  INCLUDE_DIRS include
+#  LIBRARIES waiterbot_navigation
+#  CATKIN_DEPENDS other_catkin_pkg
+#  DEPENDS system_lib
+)
+
+catkin_add_env_hooks(25.waiterbot-navigation SHELLS sh DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/env-hooks)
```
Comment (jihoonl) [inline waiterbot_suite/waiterbot_navigation/CMakeLists.txt]: Do we need env-hook for waiterbot? How does it get used?

Comment (mehditlili) [inline waiterbot_suite/waiterbot_navigation/CMakeLists.txt]: I was just confused between turtlebot simulator and the real world turtlebot. env-hooks are used in the simulator where they automatically set an environnment variable for the map_file which will be loaded at launch time. We could introduce this later for waiterbot and turtlebot_navigation packages too.

**Comment by jihoonl** [comment]
Apologies for letting you bit by bit.

How about making `custom_param_file` as arg in [amcl_demo.launch](https://github.com/turtlebot/turtlebot_apps/blob/indigo/turtlebot_navigation/launch/amcl_demo.launch)?
Then we can clean up `amcl.launch` in waiterbot_navigation like the following.

```
<launch>
  <arg custom_move_base_param "ir_params">
 <!-- Additional sensors -->
 <include file="$(find waiterbot_sensors)/launch/ir_sensors.launch"/>

 <include file="$(find turtlebot_navigation)/launch/amcl_demo.launch"/>
    <arg custom_move_base_param/>
</include>
</launch>
```

How about this way?

**Comment by jihoonl** [comment]
and also could you clean up package.xml? Then this should be ready :)

**Comment by mehditlili** [comment]
I changed the way it is passed. Now waiterbot_navigation launcher just calls amcl_demo from turtlebot and its own ir sensor launcher. It gets as parameter the custom param file and the map name.

**Comment by jihoonl** [comment]
Nice @mehditlili  Thanks!

For the last!!!(Sorry :stuck_out_tongue_closed_eyes:)

you may want to add `waiterbot_sensors` and `turtlebot_navigation` packages as run_depend in package.xml since this package is meaningless without them. :) 

Everything else looks great!

**Comment by mehditlili** [comment]
It should be fine now (finally :smile: )
I tested it on the robot and all parameters are set up correctly.

**Comment by jihoonl** [comment]
Great work! :+1: Merging it

**Pull request merged**